File: src/1046.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        while(stones.size() > 1) {
            sort(stones.begin(), stones.end());
            int wy = stones.back();
            stones.pop_back();
            int wx = stones.back();
            stones.pop_back();
            if(wy > wx) {
                stones.push_back(wy - wx);
            }
        }
        if(stones.size() == 1) {
            return stones[0];
        }else {
            return 0;
        }
    }
};

File: src/1046.py
class Solution:
    def lastStoneWeight(self, stones) -> int:
        while len(stones) > 1:
            stones.sort()
            wy = stones.pop()
            wx = stones.pop()
            if wy > wx:
                stones.append(wy - wx)
        if len(stones) == 1:
            return stones[0]
        else:
            return 0